cvmfs/network/download.cc

```diff
@@ -1688,35 +1624,76 @@ void DownloadManager::FiniHeaders() {
+                  opt_dns_server_(""),
+                  opt_timeout_proxy_(5),
+                  opt_timeout_direct_(10),
+                  opt_low_speed_limit_(1024),
+                  opt_max_retries_(0),
+                  opt_backoff_init_ms_(0),
+                  opt_backoff_max_ms_(0),
+                  enable_info_header_(false),
+                  opt_ipv4_only_(false),
+                  follow_redirects_(false),
+                  ignore_signature_failures_(false),
+                  enable_http_tracing_(false),
+                  http_tracing_headers_(vector<string>()),
+                  opt_host_chain_(NULL),
+                  opt_host_chain_rtt_(NULL),
+                  opt_host_chain_current_(0),
+                  opt_proxy_groups_(NULL),
+                  opt_proxy_groups_current_(0),
+                  opt_proxy_groups_current_burned_(0),
+                  opt_proxy_groups_fallback_(0),  // added
+                  opt_num_proxies_(0),
+                  opt_proxy_list_(""),  // added
+                  opt_proxy_fallback_list_(""),  // added
+                  opt_proxy_map_(std::map<uint32_t, ProxyInfo *>()),  // added
+                  opt_proxy_urls_(vector<string>()),  // added
+                  opt_proxy_shard_(false),
+                  sharding_policy_(SharedPtr<ShardingPolicy>()),
+                  health_check_(SharedPtr<HealthCheck>()),
+                  failover_indefinitely_(false),
+                  fqrn_(""),  // added, used in sharding policy && Interrupted()
+                  opt_ip_preference_(dns::kIpPreferSystem),
+                  proxy_template_direct_(""),  // added - should be mandatory?
+                  proxy_template_forced_(""),  // added
+                  opt_timestamp_backup_proxies_(0),
+                  opt_timestamp_failover_proxies_(0),
+                  opt_proxy_groups_reset_after_(0),
+                  opt_timestamp_backup_host_(0),
+                  opt_host_reset_after_(0),
+                  credentials_attachment_(NULL),
+                  counters_(new Counters(statistics)),
+                  ssl_certificate_store_(SslCertificateStore())  // added
 {
   atomic_init32(&multi_threaded_);
-  int retval = curl_global_init(CURL_GLOBAL_ALL);
-  assert(retval == CURLE_OK);
-  pool_handles_idle_ = new set<CURL *>;
-  pool_handles_inuse_ = new set<CURL *>;
-  pool_max_handles_ = max_pool_handles;
-  watch_fds_max_ = 4*pool_max_handles_;
-
-  opt_timeout_proxy_ = 5;
-  opt_timeout_direct_ = 10;
-  opt_low_speed_limit_ = 1024;
-  opt_proxy_groups_current_ = 0;
-  opt_proxy_groups_current_burned_ = 0;
-  opt_num_proxies_ = 0;
-  opt_proxy_shard_ = false;
-  opt_host_chain_current_ = 0;
-  opt_ip_preference_ = dns::kIpPreferSystem;
 
-  sharding_policy_ = SharedPtr<ShardingPolicy>();
-  health_check_ = SharedPtr<HealthCheck>();
-  failover_indefinitely_ = false;
+  lock_options_ =
+  reinterpret_cast<pthread_mutex_t *>(smalloc(sizeof(pthread_mutex_t)));
```

Comment:
We can use the opportunity to fix the white spacing along the way
```suggestion
    reinterpret_cast<pthread_mutex_t *>(smalloc(sizeof(pthread_mutex_t)));
```